src/index.spec.ts

```diff
@@ -0,0 +1,129 @@
+
+class SinglePage {
+    public readonly page: puppeteer.Page;
+    public messages: {
+        warning: string[]
+        log: string[]
+        info: string[]
+        debug: string[]
+        error: string[]
+    };
+    private readonly browser: puppeteer.Browser;
+    constructor(browser: puppeteer.Browser, page: puppeteer.Page) {
+        this.browser = browser;
+        this.page = page;
+        this.messages = {
+            debug: [],
+            error: [],
+            info: [],
+            log: [],
+            warning: [],
+        };
+        // collect browser console messages to be checked later
+        this.page.on('console', (consoleMessage) => {
+            const type = consoleMessage.type();
+            const text = consoleMessage.text();
+
+            switch (type) {
+                case 'debug':
+                case 'error':
+                case 'info':
+                case 'log':
+                case 'warning':
+                    this.messages[type].push(text);
+                    break;
+                default:
+                    throw new Error(`BUG: Unsupported console type: '${type}'`);
+            }
+        });
+
+        this.page.on('pageerror', async(e) => {
+            throw e;
+        });
+    }
+
+    public async close() {
+        await this.browser.close();
+    }
+}
+
+async function startBrowser(url: string) {
+
+    const puppeteerArgs = [];
+    // See https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-on-travis-ci
+    if (process.env.CI === 'true') {
+        puppeteerArgs.push('--no-sandbox');
+    }
+
+    const browser = await puppeteer.launch({
+        args: puppeteerArgs,
+        devtools: process.env.NODE_ENV === 'development',
+    });
+    const page = await browser.newPage();
+
+    const wrapper = new SinglePage(browser, page);
+    await wrapper.page.goto(url);
+
+    // Wait until React has started
+    await wrapper.page.waitForSelector('#root > h1');
+    return wrapper;
+}
+
+describe('Browser sanity tests', () => {
+
+    let devServer: http.Server | null = null;
+    let wrapper: SinglePage | null = null;
+
+    // beforeEach does not seem to work well with async functions
+    const before = async() => {
+        jest.setTimeout(5 * 60 * 1000); // dev browser takes a while to spin up
+
+        // pick a random port
+        // const port1 = (2048 + Math.round(10000 * Math.random()))
+        const port = await portfinder.getPortPromise();
+
+        devServer = http.createServer((request, response) => {
+            staticServer(request, response, {public: path.join(__dirname, '../build')});
+        });
+        devServer.listen(port);
+
+        const url = `http://localhost:${port}/`;
+        wrapper = await startBrowser(url);
+        return wrapper;
+    };
+
+    afterEach(async() => {
+        if (wrapper) {
+            await wrapper.close();
+            wrapper = null;
```

Comment:
this seems probably excessive, we can re-use the dev server and browser right? can we move that stuff to a `beforeAll` and `afterAll`? or if `before` functions don't play nicely with async we can have a helper `getBrowser` or something that re-uses that stuff internally?